### PR TITLE
TaskFinisher: use singleton SafeTimer and Finisher

### PR DESCRIPTION
**Pull request by yuyuyu101**
Fix #13938
Signed-off-by: Haomai Wang <haomai@xsky.com>

### CODE REVIEW

File: src/librbd/TaskFinisher.h [hunk 1]
```diff
@@ -16,32 +16,43 @@ class Context;
...
+  virtual ~TaskFinisherSingleton() {
     {
       Mutex::Locker l(m_lock);
       m_safe_timer->shutdown();
       delete m_safe_timer;
     }
-
     m_finisher->wait_for_empty();
     m_finisher->stop();
-    delete m_finisher;
```
Comment (dillaman) [inline src/librbd/TaskFinisher.h]: Why don't you delete the Finisher on destruction of the singleton?

**Comment by yuyuyu101** [comment]
done

**Comment by dillaman** [comment]
Testing passed -- will merge post-jewel and we can backport later.

**Comment by yuyuyu101** [comment]
@dillaman ping

**Comment by dillaman** [comment]
Test failure: http://qa-proxy.ceph.com/teuthology/jdillaman-2016-05-19_16:38:07-rbd-wip-jd-testing---basic-mira/202375/teuthology.log

**Comment by yuyuyu101** [comment]
@dillaman done, previously we lack of finisher start statement.

**Pull request merged**